src/metaswitch/clearwater/queue_manager/test/test_fvtests.py

```diff
@@ -0,0 +1,165 @@
+# Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version, along with the "Special Exception" for use of
+# the program along with SSL, set forth below. This program is distributed
+# in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details. You should have received a copy of the GNU General Public
+# License along with this program.  If not, see
+# <http://www.gnu.org/licenses/>.
+#
+# The author can be reached by email at clearwater@metaswitch.com or by
+# post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+#
+# Special Exception
+# Metaswitch Networks Ltd  grants you permission to copy, modify,
+# propagate, and distribute a work formed by combining OpenSSL with The
+# Software, or a work derivative of such a combination, even if such
+# copying, modification, propagation, or distribution would otherwise
+# violate the terms of the GPL. You must comply with the GPL in all
+# respects for all of the code used other than OpenSSL.
+# "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+# Project and licensed under the OpenSSL Licenses, or a work based on such
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+
+from time import sleep
+import logging
+import sys
+import unittest
+import json
+import os
+from functools import partial
+from metaswitch.clearwater.etcd_tests.etcdcluster import EtcdCluster
+from ..etcd_synchronizer import EtcdSynchronizer
+from plugin import TestFVPlugin
+
+logging.getLogger().addHandler(logging.StreamHandler(sys.stderr))
+logging.getLogger().setLevel(logging.INFO)
+
+def wait_for_success_or_fail(sync, pass_criteria):
+    for x in range(300):
+        val = json.loads(sync._client.read("/clearwater/site1/configuration/queue_test").value)
+        if pass_criteria(val):
+            return True
+        sleep(1)
+    print "Queue config not updated as expected, final value was: ", val
+    return False
+
+def is_queue_config_as_expected(errors, queue, val):
+    return (errors == len(val.get("ERRORED"))) and \
+           (0 == len(val.get("COMPLETED"))) and \
+           (queue == len(val.get("QUEUED")))
+
+class QueueManagerTestBase(unittest.TestCase):
+    def tearDown(self):
+        for sync in self.syncs:
+            sync.terminate()
+        sleep(5)
+        self.c.delete_datadir()
+
+    def create_synchronizers(self, count):
+        self.c = EtcdCluster(count)
+
+        self.syncs = []
+        self.plugins = []
+
+        # Create a synchronizer for each node
+        x = 0
+        for ip in self.c.servers.values():
```

Comment:
`for x, ip in enumerate(self.c.servers.values()):`

https://docs.python.org/2.3/whatsnew/section-enumerate.html